packages/image-block/src/components/Image.tsx

```diff
@@ -28,8 +20,20 @@ type ImageProps = {
     appBridge: AppBridgeBlock;
 };
 
-export const ImageComponent = ({ image, blockSettings, isEditing }: ImageProps) => {
-    const link = blockSettings.hasLink ? blockSettings.linkObject : undefined;
+export const ImageComponent = ({ image, blockSettings, isEditing, appBridge }: ImageProps) => {
+    const link = blockSettings?.hasLink && blockSettings?.linkObject?.link && blockSettings?.linkObject;
+    const imageStyle = getImageStyle(blockSettings, image.width);
+
+    const Image = (
+        <img
+            data-test-id="image-block-img"
+            className="tw-flex tw-w-full"
+            loading="lazy"
+            src={image.genericUrl.replace('{width}', `${800 * window.devicePixelRatio}`)}
```

Comment:
Is it possible that `window.devicePixelRatio` is undefined or null on some browsers/devices?

Reply:
yes indeed that could be - thanks! Added a condition also in the other blocks where it was used.